### DISCUSSION TITLE
Inaccessible Codespaces

### DISCUSSION THREAD

**Discussion by dmadisetti** [community/community#3560] [Codespaces] [answered]
Hey all,

I haven't been able to reach my codespace since the azure outage yesterday. Reading [some](https://github.com/github/feedback/discussions/3554) of the [feedback](https://github.community/t/codespace-hanging-on-start/135292): the biggest issue is less losing the space, as much as losing unpushed changes. It would be nice if we could somehow mount the codespace filesystem, even if the container was damaged.

Feel free to take a look for my particular case though: https://dmadisetti-painfulhaskell-c976.github.dev/

**Comment by jkeech** [comment, marked as answer]
@dmadisetti, sorry for the issue. Can you retry and let us know if it's working now?

**Reply by dmadisetti** [reply]
It works. Cheers. Should I leave this open or open a new ticket?

I do think accessing codespace files from is important. Especially when codespaces do become a paid product. Will github hold files hostage unless one pays for the server time?

**Reply by jkeech** [reply]
Thanks for confirming! We never want anyone to feel like their changes are trapped in the codespace. We're working on a feature to be able to push your changes to a branch at any time, which would help in this case.

Of course we _also_ want to prevent the codespace from getting stuck in a starting state in the first place, since the codespaces should always be available/reliable. Note that you do have a bit more control over the lifecycle of the codespace through the VS Code extension, where you can manually suspend it if you want (which would have worked around this specific issue). But generally, we don't want you to have to worry about the lifecycle of the codespace, and we're working on improving this scenario where the codespace was stuck in a starting state as a result of the outage yesterday.

Thanks for using the beta and providing valuable feedback for us to improve 👍

**Reply by dmadisetti** [reply]
Cool, I'll close this out since it seems like a fix is already in your pipeline. Thanks!
